legate/core/runtime.py

```diff
@@ -1504,7 +1511,9 @@ def create_store(
             shape = Shape(shape)
 
         kind = (
-            Future
+            type(data)
+            if data is not None
+            else Future
             if optimize_scalar and shape is not None and shape.volume() == 1
             else RegionField
```

Comment:
TBH I'd split this up some way, chained ternaries are a bit awkward for people and for `black`